packages/markerikson-stack-client-prototype/src/features/sources/SourceContent.tsx

```diff
@@ -0,0 +1,111 @@
+import { useMemo, useState } from "react";
+import type { HitCount, Location, PointDescription } from "@replayio/protocol";
+import { skipToken } from "@reduxjs/toolkit/dist/query";
+import CodeMirror from "@uiw/react-codemirror";
+import { javascript } from "@codemirror/lang-javascript";
+import { EditorView } from "@codemirror/view";
+
+import {
+  useGetSourceTextQuery,
+  useGetSourceHitCountsQuery,
+  useGetLineHitPointsQuery,
+  useGetPauseQuery,
+} from "../../app/api";
+import { useAppDispatch, useAppSelector } from "../../app/hooks";
+import { pointSelected } from "./sourcesSlice";
+
+export const SourceContent = () => {
+  const [selectedLocation, setSelectedLocation] = useState<Pick<
+    Location,
+    "line" | "column"
+  > | null>(null);
+  const dispatch = useAppDispatch();
+  const selectedSourceId = useAppSelector(state => state.sources.selectedSourceId);
+  const selectedPoint = useAppSelector(state => state.sources.selectedPoint);
+
+  const { currentData: sourceText } = useGetSourceTextQuery(selectedSourceId ?? skipToken);
+  const { currentData: sourceHits } = useGetSourceHitCountsQuery(selectedSourceId ?? skipToken);
```

Comment:
The concept of a `skipToken` makes sense, but I think these being hooks (and so needing to follow the rules of hooks) unfortunately makes the API a little awkward. If this data were coming from a Suspense cache, we could just conditionally read, e.g.
```js
const sourceText = selectedSourceId ? getSourceText(selectedSourceId) : null
```
This feels a little more ergonomic to me ^ but it's a small note.